src/server/node_manager.c

```diff
@@ -7555,7 +7560,35 @@ free_nodes(job *pjob)
 		for (j=0; j<psvrmom->msr_jbinxsz; j++) {
 			if (psvrmom->msr_jobindx[j] == pjob) {
 				psvrmom->msr_jobindx[j] = NULL;
+				if (is_called_by_job_purge)
+					special_case = 1;
+			}
+		}
+
+		if (special_case) {
+			sprintf(log_buffer, "\n================================================================"
+				"======================================================\nPBSPro diagnostic information."
+				" Share this log with PBSPro Team.\n=========================================="
+				"============================================================================\n"
+				"Mom's state:%lu, number of jobs on this node: %d, number of vnodes: %d",
+				psvrmom->msr_state, psvrmom->msr_numjobs, psvrmom->msr_numvnds);
+			log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_JOB, LOG_INFO, pjob->ji_qs.ji_jobid, log_buffer);
+			ch = buffer;
+			*ch = '\0';
+			for (j=0; j<psvrmom->msr_jbinxsz; j++) {
+				if (psvrmom->msr_jobindx[j] != NULL) {
+					if (ch > buffer) {
+						*ch++ = ',';
+						*ch++ = ' ';
+						*ch = '\0';
+					}
+					strcat(buffer, psvrmom->msr_jobindx[j]->ji_qs.ji_jobid);
+					ch += strlen(psvrmom->msr_jobindx[j]->ji_qs.ji_jobid);
+				}
 			}
+			sprintf(log_buffer, "Other jobs present in the node: %s\n===================================="
```

Comment:
You actually do not have to print this in this complex way. The demarcating line can actually appear in a separate log line, and that's okay